### PR TITLE
Feature/prompt for review

### PR DESCRIPTION
**Pull request by khaykov**
This PR adds Prompt for Review.

Kudos for `AppRatingDialog` code goes to @nbradbury who implemented it in Woo app for android.
I added a slight modification that uses `DialogFragment` that is retained on configuration change and does not require week reference to context.

screenshot_1550209354

**For a prompt to be displayed to user, ALL of the following conditions must be met:**

- After app has been installed for at least 7 days.
- The app should have been launched at least 10 times (lifecycle events in Android are tricky so we don't rely on this one too much)
- At least 10 significant events should have been performed by a user.

**Current significant events are:**
1. Uploading media from Media Browser.
2. Publishing a new Post or a Page.
3. Opening a Post from a reader (tapping on the post in Reader post list)
4. Checking notification details.

I wanted to start slowly and carefully, so we only have 4 events for now. After we get some data on the usage of the prompt we can add more events.

**What we are tracking:**

- app_reviews_saw_prompt
- app_reviews_cancelled_prompt
- app_reviews_rated_app
- app_reviews_declined_to_rate_apt
- app_reviews_decided_to_rate_later

I would like to get design feedback on copy, significant events and tracking events 🙇

### CODE REVIEW

**Comment by nbradbury** [comment]
> I added a slight modification that uses `DialogFragment`

👍

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java [hunk 1]
```diff
@@ -315,6 +316,9 @@ public void run() {
             GCMRegistrationIntentService.enqueueWork(this,
                     new Intent(this, GCMRegistrationIntentService.class));
         }
+
+        AppRatingDialog.INSTANCE.init(this);
+        AppRatingDialog.INSTANCE.showRateDialogIfNeeded(getFragmentManager());
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: We probably don't want to show the rating dialog if the logic above caused the login or signup epilogues to appear. I also wonder if we should skip it if the user opened the app from a push notification or a shortcut. The reasoning is that in those situations the user had a specific task in mind and may be annoyed at being interrupted by a rating dialog.

**Comment by nbradbury** [comment]
@khaykov I'm done with my first round. Code looks good - just had one comment about skipping the dialog in certain cases.

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java (hunk 1 above)
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: Good idea! There is a lot of stuff going on in `onCreate`, so I updated the code to only show the dialog when the user launches app "normally".

**Comment by jtreanor** [comment]
I have updated the base of this branch to `release/11.9` since the code freeze has now happened.

**Comment by khaykov** [comment]
Added tracking of significant events: 

| Event name                                | Parameter Name | Parameter Value         |
|-------------------------------------------|----------------|-------------------------|
| app_reviews_significant_event_incremented | source         | media_upload            |
| app_reviews_significant_event_incremented | source         | notification_details    |
| app_reviews_significant_event_incremented | source         | publishing_post_or_page |
| app_reviews_significant_event_incremented | source         | opening_reader_post     |

**Comment by SylvesterWilmott** [comment]
Tracking looks good @khaykov, thanks.

One questions, once I meet the requirements for showing this dialog when exactly will I see it? Is it on app launch or when the final significant event has been complete?

**Comment by khaykov** [comment]
@SylvesterWilmott You will see the dialog next time you launch the app.

**Comment by jtreanor** [comment]
@khaykov I have updated to base back to `develop` since you changed the milestone to 12.0.

**Comment by SylvesterWilmott** [comment]
This looks good @khaykov !

**Comment by nbradbury** [comment]
Looks good! :shipit:

**Pull request merged**